submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,76 @@
+# Cable Driver Enhancement - IP-in-IP
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional enhancements needed for the Submariner Cable Driver
+to support of IP-in-IP for unencrypted connections between clusters or for use as an overlay with
+IPSec in transport mode for encrypted connections between clusters.
+
+For IP in IP, the benefits fall into the following categories:
+
+* **Performance**: IP in IP is has a lower overhead than VXLAN which could lead to performance
+improvements in terms of packet overhead (bandwidth consumption) as well as the cost of processing
+packets. Some of the performance improvements are shown below [1]:
+
+  * ![Throughput rate vs Kernel](./images/tx-tput-rate.png)
+  * ![Throughput rate Gbps](./images/tx-tput-gbps.png)
+  * ![CPU cycles](./images/cpu-cycles.png)
+
+* **Interoperability**: IP in IP can be used to transport packets between domains when the protocol
+in those domains (e.g. IPv6) is not supported by intermediary networks (e.g. IPv4).
+
+## Proposal
+
+This enhancement proposes to support IP-in-IP for un-encrypted connection between clusters. A
+new IP tunnelling interface `ipip-tunnel` will be added to the active gateway node of each cluster
+and it will act as the Tunnel End Point. The traffic which is destined to other clusters will be
+forwarded via this interface to the appropriate clusters.
+
+> **_NOTE:_** A single IP Tunnelling interface will be added, and tunnels managed using
+`# ip route add encap ip <args>` in order to avoid a point to point connection per tunnel.
+
+### Design Details
+
+* A new interface "ipip-tunnel" will be added.
+* The IP tunnelling interface will be one-to-many, that is a single interface will be used to
+connect to all the joined clusters.
+* For the "ipip-tunnel" IP, the prefix "243" will be used, followed by the rest of the blocks
+from the endpoint private IP address. For example, if the private IP address is 10.2.96.1 the
+"ipip-tunnel" IP would be 243.2.96.1.
```

Comment:
Is there some importance to this decision?

Reply:
>Is there some importance to this decision?

Since we do not have a global IPAM to allocate unique ips to these interface, we took this approach.

Reply:
SO the address selection should be clearly documented, eg is address randomized or is it expected the user using the installer supply it explicitly, or is it taken from a range in the broker, and if so how do we make sure two endpoints dont get the same address